subprojects/testing-base/src/testFixtures/groovy/org/gradle/testing/AbstractTestFrameworkIntegrationTest.groovy

```diff
@@ -150,11 +157,114 @@ abstract class AbstractTestFrameworkIntegrationTest extends AbstractIntegrationS
         testResult.assertTestClassesExecuted('SomeTest', 'NewTest')
     }
 
+    def "honors test case filter from --tests flag"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.$passingTestCaseName"
+
+        then:
+        testResult.assertTestClassesExecuted('SomeOtherTest')
+        testResult.testClass('SomeOtherTest').assertTestPassed(passingTestCaseName)
+    }
+
+    def "honors test suite filter from --tests flag"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        run testTaskName, '--tests', "${testSuite('SomeOtherTest')}.*"
+
+        then:
+        testResult.assertTestClassesExecuted('SomeOtherTest')
+        testResult.testClass('SomeOtherTest').assertTestPassed(passingTestCaseName)
+    }
+
+    def "reports when no matching methods found"() {
+        given:
+        createPassingFailingTest()
+
+        //by command line
+        when: fails(testTaskName, "--tests", "${testSuite('SomeTest')}.missingMethod")
+        then: failure.assertHasCause("No tests found for given includes: [${testSuite('SomeTest')}.missingMethod](--tests filter)")
+
+        //by build script
+        when:
+        buildFile << "${testTaskName}.filter.includeTestsMatching '${testSuite('SomeTest')}.missingMethod'"
+        fails(testTaskName)
+        then: failure.assertHasCause("No tests found for given includes: [${testSuite('SomeTest')}.missingMethod](filter.includeTestsMatching)")
+    }
+
+    def "task is out of date when --tests argument changes"() {
+        String testTaskPath = ":$testTaskName"
+
+        given:
+        createPassingFailingTest()
+
+        when: run(testTaskName, "--tests", "${testSuite('SomeOtherTest')}.$passingTestCaseName")
+        then: testResult.testClass("SomeOtherTest").assertTestsExecuted(passingTestCaseName)
```

Comment:
nit picky thing, can these be on separate lines from the labels?

Reply:
Sure, done.